langchain4j-jlama/src/main/java/dev/langchain4j/model/jlama/JacksonJsonCodec.java

```diff
@@ -0,0 +1,84 @@
+package dev.langchain4j.model.jlama;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+
+class JacksonJsonCodec {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .enable(SerializationFeature.INDENT_OUTPUT)
+            .registerModule(new JavaTimeModule())
+            .registerModule(new SimpleModule()
+                    .addSerializer(LocalDate.class, new LocalDateSerializer())
+                    .addDeserializer(LocalDate.class, new LocalDateDeserializer())
+                    .addSerializer(LocalTime.class, new LocalTimeSerializer())
+                    .addDeserializer(LocalTime.class, new LocalTimeDeserializer())
+                    .addSerializer(LocalDateTime.class, new LocalDateTimeSerializer())
+                    .addDeserializer(LocalDateTime.class, new LocalDateTimeDeserializer())
+            );
+
+    public static <T> T fromJson(String json, Class<T> type) {
+        try {
+            if (Map.class.isAssignableFrom(type)) {
+                return OBJECT_MAPPER.readValue(json, OBJECT_MAPPER.getTypeFactory().constructMapType(Map.class, String.class, String.class));
```

Comment:
why is this special treatment required?
shouldn't it be `Map<String, Object>`?

Reply:
Hi @langchain4j 
I referred to the code present in the GsonJsonCodec class and in the fromJSON function I saw the docsting

 /**
     * Reads a JSON string and returns an object of the specified type.
     *
     * <p>As a special case, if the type is {@link Map}, the returned object is
     * parsed as a {@code Map<String, String>}.
     *
     * @param json JSON string to be parsed.
     * @param type Type of the object to be returned.
     * @return the object represented by the JSON string.
     * @param <T> Type of the object to be returned.
     */
     
    @Override
    public <T> T fromJson(String json, Class<T> type) {
        return GSON.fromJson(json, type);
    }
    
    So, to maintain compatibility I made it to rerurn Map<String, Object>

Reply:
Will be modifying as per Map<String, Object>

Reply:
Sorry about that, apparently javadoc comment was outdated

Reply:
No Worries :)